src/printer.js

```diff
@@ -1934,8 +1934,16 @@ function genericPrintNoParens(path, options, print, args) {
             tabWidth
           );
 
+          const parent = path.getParentNode(1);
+          const shouldBreak = !(
+            parent &&
+            parent.type === "NewExpression" &&
+            parent.callee.type === "Identifier" &&
+            parent.callee.name === "RegExp"
+          );
+
           const aligned = addAlignmentToDoc(
-            expressions[i],
+            shouldBreak ? expressions[i] : docUtils.removeLines(expressions[i]),
```

Comment:
We've gotten bitten by using removeLines in the past, because it will remove lines in nested nodes, eg functions, arrays, etc. I don't think this is a problem for template strings in RegExp literals, but it's worth being aware of.

Reply:
Yeah, please don't use remove lines. What you want to do instead is to reprint the few patterns you identified here.

You can start with member expressions with only identifiers and member expressions and no comments.